packages/eslint-plugin/src/rules/space-before-blocks.ts

```diff
@@ -0,0 +1,102 @@
+import { TSESTree } from '@typescript-eslint/experimental-utils';
+import { getESLintCoreRule } from '../util/getESLintCoreRule';
+import * as util from '../util';
+
+const baseRule = getESLintCoreRule('space-before-blocks');
+
+export type Options = util.InferOptionsTypeFromRule<typeof baseRule>;
+export type MessageIds = util.InferMessageIdsTypeFromRule<typeof baseRule>;
+
+export default util.createRule<Options, MessageIds>({
+  name: 'space-before-blocks',
+  meta: {
+    type: 'layout',
+    docs: {
+      description: 'Enforces consistent spacing before blocks',
+      recommended: false,
+      extendsBaseRule: true,
+    },
+    fixable: baseRule.meta.fixable,
+    hasSuggestions: baseRule.meta.hasSuggestions,
+    schema: baseRule.meta.schema,
+    messages: {
+      // @ts-expect-error -- we report on this messageId so we need to ensure it's there in case ESLint changes in future
+      unexpectedSpace: 'Unexpected space before opening brace.',
+      // @ts-expect-error -- we report on this messageId so we need to ensure it's there in case ESLint changes in future
```

Comment:
This would have been a side-effect of #3844.
We could definitely improve `getESLintCodeRule` to make these mutable!

---

This is an unfortunate problem that arrises because we support ESLint v6. Many rules didn't have `messages` defined until ESLint v7 (when myself and another contributor added them to all rules).